From 7996cfe15ee97c5dfe4adbcabe0790e37df5e54c Mon Sep 17 00:00:00 2001
From: Yifeng Zhao <zyf@rock-chips.com>
Date: Wed, 4 Sep 2019 09:12:18 +0800
Subject: [PATCH] drivers: rk_nand: add disk information to kernel list

set nand devices path to /dev/platform/xxxxxxx.nandc/rknand0
for android 9.0 and android 10 while using gpt.

Change-Id: I21c5d71ee6ed5d8520630174329c6da1c2764430
Signed-off-by: Yifeng Zhao <zyf@rock-chips.com>
---
 drivers/rk_nand/rk_nand_blk.c | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/drivers/rk_nand/rk_nand_blk.c b/drivers/rk_nand/rk_nand_blk.c
index 16cc54cf608d..225e775e7530 100644
--- a/drivers/rk_nand/rk_nand_blk.c
+++ b/drivers/rk_nand/rk_nand_blk.c
@@ -613,7 +613,10 @@ static int nand_add_dev(struct nand_blk_ops *nandr, struct nand_part *part)
 	if (dev->readonly)
 		set_disk_ro(gd, 1);
 
-	add_disk(gd);
+	if (gd->flags != GENHD_FL_EXT_DEVT)
+		add_disk(gd);
+	else
+		device_add_disk(g_nand_device, gd);
 
 	return 0;
 }
-- 
2.35.3

